frontend/src/pages/modelServing/screens/projects/InferenceServiceModal/ManageInferenceServiceModal.tsx

```diff
@@ -46,12 +46,12 @@ const ManageInferenceServiceModal: React.FC<ManageInferenceServiceModalProps> =
     translateDisplayNameForK8s(createData.name).length <= 253;
 
   React.useEffect(() => {
-    if (projectContext) {
+    if (projectContext && !createData.project && !createData.servingRuntimeName) {
       const { currentProject, currentServingRuntime } = projectContext;
       setCreateData('project', currentProject.metadata.name);
       setCreateData('servingRuntimeName', currentServingRuntime?.metadata.name || '');
     }
-  }, [projectContext, setCreateData]);
+  }, [projectContext, setCreateData, createData.servingRuntimeName, createData.project]);
```

Comment:
You should remove `projectContext` from the dependencies here because in every instance of the caller of this component, the `projectContext` is a new object. 
First assign the values of interest to variables and then use only those in the `useEffect`

Reply:
Yes, you are completly right